source/lib/rocprof-sys/library/amd_smi.cpp

```diff
@@ -184,18 +185,59 @@ data::sample(uint32_t _dev_id)
 #endif
     ROCPROFSYS_AMDSMI_GET(get_settings(m_dev_id).mem_usage, amdsmi_get_gpu_memory_usage,
                           sample_handle, AMDSMI_MEM_TYPE_VRAM, &m_mem_usage);
-    ROCPROFSYS_AMDSMI_GET(get_settings(m_dev_id).vcn_activity,
-                          amdsmi_get_gpu_metrics_info, sample_handle, &_gpu_metrics);
-    ROCPROFSYS_AMDSMI_GET(get_settings(m_dev_id).jpeg_activity,
-                          amdsmi_get_gpu_metrics_info, sample_handle, &_gpu_metrics);
+    is_vcn_or_jpeg_activity_enabled =
+        get_settings(m_dev_id).vcn_activity || get_settings(m_dev_id).jpeg_activity;
+    ROCPROFSYS_AMDSMI_GET(is_vcn_or_jpeg_activity_enabled, amdsmi_get_gpu_metrics_info,
+                          sample_handle, &_gpu_metrics);
 
-    for(const auto& v_activity : _gpu_metrics.vcn_activity)
+    // Process metrics if either VCN or JPEG activity is enabled
+    if(is_vcn_or_jpeg_activity_enabled)
     {
-        if(v_activity != UINT16_MAX) m_vcn_metrics.push_back(v_activity);
-    }
-    for(const auto& j_activity : _gpu_metrics.jpeg_activity)
-    {
-        if(j_activity != UINT16_MAX) m_jpeg_metrics.push_back(j_activity);
+        // Helper lambda to fill busy metrics from a source array
+        auto fill_busy_metrics = [](auto& dest, const auto& src) {
+            for(const auto& val : src)
+            {
+                if(val != UINT16_MAX) dest.push_back(val);
+            }
+        };
+
+        if(gpu::is_vcn_activity_supported(_dev_id) &&
+           gpu::is_jpeg_activity_supported(_dev_id))
```

Comment:
```suggestion
        if(gpu::is_vcn_activity_supported(m_dev_id) &&
           gpu::is_jpeg_activity_supported(m_dev_id))
```
Above we use `m_dev_id`, but here we're using `_dev_id`.  Was there a reason to used one over the other? If not, let's use `m_dev_id` for consistency.

Reply:
oversight again :-( Fixing this.